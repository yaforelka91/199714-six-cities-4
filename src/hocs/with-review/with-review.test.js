import React from 'react';
import renderer from 'react-test-renderer';
import {PropTypes} from 'prop-types';
import withReview from './with-review.js';

const MockComponent = ({renderRating, renderTextarea}) => {
  return (
    <div>
      {renderRating()}
      {renderTextarea()}
    </div>
  );
};

MockComponent.propTypes = {
  renderRating: PropTypes.func.isRequired,
  renderTextarea: PropTypes.func.isRequired,
};

const MockComponentWrapped = withReview(MockComponent);

describe(`withReviewSnapshot`, () => {
  it(`should render withReview`, () => {
    const tree = renderer.create(
        <MockComponentWrapped
          onReviewFormSubmit={() => {}}
          offerId={-1}
        />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
