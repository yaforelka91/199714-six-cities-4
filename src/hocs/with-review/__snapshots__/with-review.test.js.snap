// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withReviewSnapshot should render withReview 1`] = `
<div>
  <div
    className="reviews__rating-form form__rating"
  >
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="5-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="5"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="5-stars"
      title="perfect"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="4-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="4"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="4-stars"
      title="good"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="3-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="3"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="3-stars"
      title="not bad"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="2-stars"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="2"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="2-stars"
      title="badly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
    <input
      checked={false}
      className="form__rating-input visually-hidden"
      id="1-star"
      name="rating"
      onChange={[Function]}
      type="radio"
      value="1"
    />
    <label
      className="reviews__rating-label form__rating-label"
      htmlFor="1-star"
      title="terribly"
    >
      <svg
        className="form__star-image"
        height="33"
        width="37"
      >
        <use
          xlinkHref="#icon-star"
        />
      </svg>
    </label>
  </div>
  <textarea
    className="reviews__textarea form__textarea"
    id="review"
    maxLength="300"
    name="review"
    onChange={[Function]}
    placeholder="Tell how was your stay, what you like and what can be improved"
    value=""
  />
</div>
`;
